_source/_posts/2021-10-27-k8s-api-server-oidc.md

```diff
@@ -0,0 +1,551 @@
+
+Since the API server is the only part of the Kubernetes cluster that a client can access, it is essential to secure the API server. Unauthorized access of the API server can lead to the hijack of the entire cluster and maybe even your infrastructure, data theft, and so on. Hence, configuring users and roles properly is a must to secure the cluster, especially i9n organizations where more than one user can access the cluster.
+
+# Why OpenID Connect?
+
+> [OpenID Connect](https://developer.okta.com/docs/concepts/oauth-openid/) is an authentication standard built on top of OAuth 2.0. It adds an additional token called an ID token. OpenID Connect also standardizes areas that OAuth 2.0 leaves up to choices, such as scopes, endpoint discovery, and dynamic registration of clients
+
+While the default authentication mechanism of the Kubernetes API server might be enough for simple use cases, where only a handful of people manages the cluster, it does not scale for bigger organizations. It's definitely not the most secure way as well. This is because Kubernetes does not handle user management and expects this to be done outside. This is where OpenID Connect (OIDC) comes in. OIDC can manage users and groups, which works very well with the Kubernetes RBAC to provide very granular control of who can access what inside a cluster.
+
+{% img blog/k8s-api-server-oidc/kube-login-oidc.jpg alt:"K8s OIDC flow" width:"800" %}{: .center-image }
+
+Having an OIDC integration also means you can use the same OIDC provider used to do SSO in your existing infrastructure to access your Kubernetes cluster as well, like Okta or Keycloak, for example.
+
+# Using Okta as an OIDC provider to secure the API server.
+
+> Okta, Inc. is an identity and access management company providing cloud software that helps companies manage and secure user authentication into applications and for developers to build identity controls into applications, websites, web services, and devices. Okta is a certified OpenID Connect provider.
+
+Let us see how we can secure the Kubernetes API server using Okta as an OIDC provider and use RBAC to control access right from the Okta admin console. If you are using Amazon EKS, then check [this](/blog/2021/10/08/secure-access-to-aws-eks) specific tutorial for using Okta OIDC with EKS.
+
+## What You'll Need to Get Started
+
+Before you start trying this out, make sure you have access to the following.
+
+- An Okta account. You can sign up for a free account [here](https://www.okta.com/free-trial/). If you like, you can use another OIDC provider or [Dex](https://github.com/dexidp/dex), and the steps should be similar.
```

Comment:
the /free-trial/ link is for workforce identity. If it's on the developer blog, it typically requires API Access Management, which doesn't come with workforce for free. It does come with developer accounts for free, but you need to sign up at developer.okta.com/signup.

Reply:
Good to know. I'll update